### PR TITLE
Preserve -L -l pairings fetched from external deps

### PR DESCRIPTION
**Pull request by nirbheek**
While adding link args for external deps, sometimes different libraries come from different prefixes, and an older version of the same library might be present in other prefixes and we don't want to accidentally pick that up.

For example:

```
/usr/local/lib/libglib-2.0.so
/usr/local/lib/pkgconfig/glib-2.0.pc
/usr/local/lib/libz.so
/usr/local/lib/pkgconfig/zlib.pc
/home/mesonuser/.local/lib/libglib-2.0.so
/home/mesonuser/.local/lib/pkgconfig/glib-2.0.pc

PKG_CONFIG_PATH="/home/mesonuser/.local/lib/pkgconfig/:/usr/local/lib/pkgconfig/"
```

If a target uses `dependencies : [glib_dep, zlib_dep]`, it will end up
using `/usr/local/lib/libglib-2.0.so` instead of `/home/mesonuser/.local/lib/libglib-2.0.so` despite using the pkg-config file in `/home/mesonuser/.local/lib/pkgconfig` because we reorder the `-L`
flag and separate it from the `-l` flag.

With this change, external link arguments will be added to the compiler list without de-dup or reordering.

Closes https://github.com/mesonbuild/meson/issues/1718

### CODE REVIEW

**Comment by nirbheek** [comment]
I will add a test once @lantw44 confirms whether this fixes the issue for him.

**Comment by jpakkane** [comment]
It would be great if this had a test with an actual project in addition to the unit test.

**Comment by nirbheek** [comment]
We can probably reuse the test from #1928?

**Comment by jpakkane** [comment]
Go for it. It should also retrigger OSX, whose failure is unrelated.

**Comment by nirbheek** [comment]
macOS builds are backlogged, I don't know if that will clear up tonight.

**Comment by jpakkane** [comment]
I cancelled all ones not related to this MR.

File: run_unittests.py [hunk 1]
```diff
@@ -1745,6 +1755,25 @@ def test_run_installed(self):
         env['LD_LIBRARY_PATH'] = installed_libdir
         self.assertEqual(subprocess.call(installed_exe, env=env), 0)
 
+    def test_order_of_l_arguments(self):
+        testdir = os.path.join(self.unit_test_dir, '9 -L -l order')
+        os.environ['PKG_CONFIG_PATH'] = testdir
+        self.init(testdir)
+        # NOTE: .pc file has -Lfoo -lfoo -Lbar -lbar but pkg-config reorders
+        # the flags before returning them to -Lfoo -Lbar -lfoo -lbar
+        expected_order = ['-L/me/first', '-L/me/second','-lfoo1', '-lfoo2',
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[flake8]__
* [E231] missing whitespace after ','

**Comment by jpakkane** [comment]
OSX is still broken in Travis but I tested that it works on my MBPro laptop.

**Comment by lantw44** [comment]
This patch makes meson behave like automake, so yes, it works.

**Pull request merged**